chapter2/tests/t19/terraform/_modules/container/main.tf

```diff
@@ -0,0 +1,21 @@
+resource "docker_image" "container_name" {
+  name         = var.container_image
+  keep_locally = var.container_image_keep_locally
+}
+
+resource "docker_container" "container_name" {
+  image = docker_image.container_name.image_id
+  name  = var.container_name
+  ports {
+    internal = var.container_ports.internal
+    external = var.container_ports.external
+  }
+  dynamic "volumes" {
+    for_each = var.container_volumes
+    content {
+      host_path      = abspath("${path.cwd}/${volumes.value.host}")
```

Comment:
> This is vary specific for `nginx` module, you should just use vars for everything inside this `content` block, because it is a base container module, here you make all volumes always `read only`, which is incorrect

@kseniyashaydurova 
Done